### PR TITLE
ci: Added workflow for ESCU test

### PR DESCRIPTION
**Pull request by rajkarkar-crest**

### CODE REVIEW

**Review by bhargav-nariyani-crest — APPROVED** [review]

**Review by uoboda-splunk — APPROVED** [review]

File: enforce/.github/workflows/escu-manual-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,244 @@
+name: escu-manual-workflow
```
Comment (artemrys) [inline enforce/.github/workflows/escu-manual-workflow.yml:1]: Can we add it to the reusable workflow?

File: enforce/.github/workflows/escu-manual-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+      matrix_supportedSplunk: ${{ steps.matrix.outputs.latestSplunk }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: false
+          persist-credentials: false
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
```
Comment (artemrys) [inline enforce/.github/workflows/escu-manual-workflow.yml:25]: I think we don't need this anymore.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -14,13 +14,13 @@ repos:
       - id: check-merge-conflict
       - id: debug-statements
       - id: trailing-whitespace
-        exclude: ^(tests/requirement_test/logs)
+        exclude: ^(tests/requirement_test/logs|tests/knowledge/samples)
```
Comment (artemrys) [inline .pre-commit-config.yaml:17]: Please add `tests/escu/.escu_detections` here as well.

**Review by artemrys — APPROVED** [review]

File: enforce/.github/workflows/escu-manual-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,244 @@
+name: escu-manual-workflow
+on:
+  workflow_dispatch:
+    inputs:
+      TA_BUILD:
+        description: 'TA build number (e.g. s3://ta-production-artifacts/ta-apps/{ta-name}-$build_number$.spl)'
+        required: true
+      TESTS:
+        description: 'Comma-Separated List of detections to run'
```
Comment (artemrys) [inline enforce/.github/workflows/escu-manual-workflow.yml:9]: What are the expected values here?

**Review by artemrys — CHANGES_REQUESTED** [review]

File: enforce/.github/workflows/escu-manual-workflow.yml (hunk 1 above)
Comment (rajkarkar-crest) [reply enforce/.github/workflows/escu-manual-workflow.yml:1]: @artemrys We have already added this test suite to the reusable workflow(https://github.com/splunk/addonfactory-workflow-addon-release/pull/93), but this is a separate workflow that lets the dev team execute this test-suite manually.

File: enforce/.github/workflows/escu-manual-workflow.yml (hunk 4 above)
Comment (rajkarkar-crest) [reply enforce/.github/workflows/escu-manual-workflow.yml]: The Dev team has a detection list applicable for the particular add-on. They need to pass the comma-separated list of detection to run. Added example for reference.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (rajkarkar-crest) [reply .pre-commit-config.yaml]: Added `tests/escu/.escu_detections`.

File: enforce/.github/workflows/escu-manual-workflow.yml (hunk 2 above)
Comment (rajkarkar-crest) [reply enforce/.github/workflows/escu-manual-workflow.yml]: Removing this.

File: enforce/.github/workflows/escu-manual-workflow.yml (hunk 1 above)
Comment (artemrys) [reply enforce/.github/workflows/escu-manual-workflow.yml:1]: I meant that you can add something like this https://github.com/splunk/addonfactory-workflow-addon-release/blob/main/.github/workflows/reusable-build-test-release.yml, but specifically for the ESCU manual run.

The reason for this is that if we need to change something, we need to push a new change from the template which will trigger a lot of PRs which need to be merged which takes time and unnecessary maintenance.

Comment (rajkarkar-crest) [reply enforce/.github/workflows/escu-manual-workflow.yml:1]: @artemrys Agreed. As per your suggestion, I have added workflow in addonfactory-workflow-addon-release which can be called from addons. I raised a PR https://github.com/splunk/addonfactory-workflow-addon-release/pull/95. Can you please review it?

Comment (rajkarkar-crest) [reply enforce/.github/workflows/escu-manual-workflow.yml:1]: Also updated changes escu-manual-workflow to call workflow from https://github.com/splunk/addonfactory-workflow-addon-release.

File: enforce/.github/workflows/escu-manual-workflow.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      TA_BUILD:
+        description: 'TA build number (e.g. s3://ta-production-artifacts/ta-apps/{ta-name}-$build_number$.spl)'
+        required: true
+      TESTS:
+        description: 'Comma-Separated List of detections to run (e.g. detection1,detection2,detection3)'
+        required: true
+
+jobs:
+  call-workflow:
+    uses: splunk/addonfactory-workflow-addon-release/.github/workflows/reusable-escu-manual-workflow.yml@v2.1
```
Comment (uoboda-splunk) [inline enforce/.github/workflows/escu-manual-workflow.yml:14]: It must be a separate repository for `reusable-escu-manual-workflow` We will have a mismatch of versions with `build-test-release` all the time

**Review by uoboda-splunk — CHANGES_REQUESTED** [review]

File: enforce/.github/workflows/escu-manual-workflow.yml (hunk 5 above)
Comment (uoboda-splunk) [reply enforce/.github/workflows/escu-manual-workflow.yml:14]: Agreed to do it as follow up task

**Review by uoboda-splunk — APPROVED** [review]

**Review by bhargav-nariyani-crest — APPROVED** [review]

File: enforce/.github/workflows/escu-manual-workflow.yml (hunk 5 above)
Comment (rajkarkar-crest) [reply enforce/.github/workflows/escu-manual-workflow.yml:14]: Reference Jira: https://github.com/splunk/addonfactory-repository-template/pull/277

**Review by artemrys — APPROVED** [review]

**Pull request merged**